libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxDistanceFormatter.kt

```diff
@@ -8,69 +8,64 @@ import android.text.style.RelativeSizeSpan
-import com.mapbox.turf.TurfConstants.UNIT_FEET
-import com.mapbox.turf.TurfConstants.UNIT_KILOMETERS
-import com.mapbox.turf.TurfConstants.UNIT_METERS
-import com.mapbox.turf.TurfConstants.UNIT_MILES
+import com.mapbox.turf.TurfConstants
 import com.mapbox.turf.TurfConversion
 import java.text.NumberFormat
 import java.util.Locale
+import kotlin.math.roundToInt
 
-/**
- * Creates an instance of DistanceFormatter, which can format distances in meters
- * based on a language format and unit type.
- *
- *
- * This constructor will infer device language and unit type using the device locale.
- *
- * @param language for which language
- * @param unitType to use, or UNDEFINED to use default for locale country
- * @param context from which to get localized strings from
- * @param roundingIncrement increment by which to round small distances
- */
-class MapboxDistanceFormatter(
-    private val context: Context,
-    language: String?,
-    @VoiceUnit unitType: String,
-    @RoundingIncrement private val roundingIncrement: Int
-) : DistanceFormatter {
-
-    private val unitStrings = hashMapOf<String, String>(
-        UNIT_KILOMETERS to context.getString(R.string.kilometers),
-        UNIT_METERS to context.getString(R.string.meters),
-        UNIT_MILES to context.getString(R.string.miles),
-        UNIT_FEET to context.getString(R.string.feet)
-    )
-    private val largeUnit = when (IMPERIAL == unitType) {
-        true -> UNIT_MILES
-        false -> UNIT_KILOMETERS
+class MapboxDistanceFormatter(private val context: Context, private val unitType: String, private val roundingIncrement: Int) : DistanceFormatter {
+
+    // These values are in meters.
+    private val smallDistanceUpperThreshold = 400.0
+    private val mediumDistanceLowerThreshold = 401.0
+    private val mediumDistanceUpperThreshold = 10000.0
+
+    private val smallUnit = when (unitType) {
+        IMPERIAL -> TurfConstants.UNIT_FEET
+        else -> TurfConstants.UNIT_METERS
     }
-    private val smallUnit = when (IMPERIAL == unitType) {
-        true -> UNIT_FEET
-        false -> UNIT_METERS
+
+    private val largeUnit = when (unitType) {
+        IMPERIAL -> TurfConstants.UNIT_MILES
+        else -> TurfConstants.UNIT_KILOMETERS
     }
-    private val language: String
-    @VoiceUnit
-    private val unitType: String
-    private val numberFormat: NumberFormat
-
-    init {
-        val locale = when (language == null) {
-            true -> context.inferDeviceLocale()
-            false -> Locale(language)
-        }
-        this.language = locale.language
-        this.unitType = when (IMPERIAL != unitType && METRIC != unitType) {
-            true -> context.inferDeviceLocale().getUnitTypeForLocale()
-            false -> unitType
+
+    private val locale: Locale by lazy {
+        context.inferDeviceLocale()
+    }
+
+    companion object {
+        @JvmStatic
+        fun builder(context: Context): Builder = Builder(context)
+    }
+
```

Comment:
Do we want to expose a `toBuilder`? 🤔

Reply:
I was following an example in classes like NavigationRoute and CurrentLegAnnotation. There doesn't appear to be a consistency in the codebase.